### DISCUSSION TITLE
Apache HTACCESS config for Vvveb CMS installed in subdirectory of another PHP application (Wordpress)

### DISCUSSION THREAD

**Discussion by HustleTrader** [givanz/Vvveb#23] [General]
Having some real troubles getting the CMS installed in a subdirectory of another PHP application (Wordpress in root folder)

Root DIR (Wordpress): https://dev.website.com
Sub DIR (Vvveb): https://dev.website.com/cms

I've managed to get most of the CMS to work, by editing the htaccess file and index.php file in a few different locations. The index.php file references the public path and public theme path, which i believe is where i'm having some troubles. Certain things like public admin theme and public site themes (scripts and images) are still returning the root directory 404 page for certain files in the public folder.

Im really lost on this issue, and if you have any recommendations for setting a base URL in the application config or something that i'm missing, I would really appreciate the help. Apache and NGINX configurations really aren't my strong suit.

I'm VERY impressed with Vvveb and everything you've done with the editor and would love to help develop an asynchronous NodeJS version of this CMS, and help to write some new logic to integrate with 3rd party or private APIs, and even headless CMS content. Even better than integrating with a headless CMS for content would be to create some kind of UI to scaffold content types in Vvveb and open API routes to access/modify them directly from this CMS.

Thank you for all you've done so far and I look forward to contributing on this project, because it hits close to home for me, and I appreciate any and all help you can provide on this issue.

**Comment by givanz** [comment]
Hi

Thank you.

I added support for subdirectory install in the last zip update.
You need to edit `/index.php` and `admin/index.php` and set `V_SUBDIR_INSTALL` as the subdirectory name, for example if installed in `/vvveb` folder you need to set it to

```php
defined('V_SUBDIR_INSTALL') || define('V_SUBDIR_INSTALL', '/vvveb');
```

Installing in a Wordpress installation folder might cause some blog posts urls to reach Wordpress instead of Vvveb you might need to tweak Wordpress .htaccess or change Vvveb posts/product url routes in config/routes.php

After changing to subdirectory all urls will work with the new path you only need to edit menu entries and update the urls for `/vvveb/blog` `/vvveb/page/contact` etc
You will also need to clear cache for new urls to be generated.

Running Vvveb from / or /subdirectory should be done for testing only, the recommended way is to set document root to /public to avoid exposing php files to public and reduce security risks.

Vvveb Rest API which is not implemented yet will allow full access to all admin controllers exposing all functionality through the API.
